src/components/WeatherIndex.js

```diff
@@ -0,0 +1,113 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useEffect, useState } from 'react';
+import {
+  Sparklines,
+  SparklinesLine,
+  SparklinesReferenceLine,
+} from 'react-sparklines';
+import { fetchWeather } from '../actions';
+
+const WeatherIndex = () => {
+  const dispatch = useDispatch();
+  const [city, setCity] = useState('');
+
+  const renderWeather = async () => {
+    const { type, payload } = await fetchWeather(city);
```

Comment:
This function is poorly named as it does not render anything.

Also, this shouldn't be in the component. Use Redux Middleware.